addons/project/models/project.py

```diff
@@ -1417,6 +1420,42 @@ def _compute_stage_id(self):
             else:
                 task.stage_id = False
 
+    @api.depends('user_ids')
+    def _compute_portal_user_names(self):
+        """ This compute method allows to see all the names of assigned users to each task contained in `self`.
+
+            When we are in the project sharing feature, the `user_ids` contains only the users if we are a portal user.
+            That is, only the users in the same company of the current user.
+            So this compute method is a related of `user_ids.name` but with more records that the portal user
+            can normally see.
+            (In other words, this compute is only used in project sharing views to see all assignees for each task)
+        """
+        if not self.ids:  # Then the compute is triggered for a new record in the form view for instance.
+            for task in self:
+                task.portal_user_names = ', '.join(task.user_ids.mapped('name'))
+            return
+
+        users_by_task_read_group = self.search_read([('id', 'in', self.ids)], ['id', 'user_ids'])
+        assignee_ids = {res['user_ids'][index] for res in users_by_task_read_group for index in range(len(res['user_ids']))}
+        assignees_read = self.env['res.users'].search_read([('id', 'in', list(assignee_ids))], ['id', 'name'])
+        assignees_dict = {res['id']: res['name'] for res in assignees_read}
+        assignees_name_by_task = {res['id']: [username for user_id, username in assignees_dict.items() if user_id in res['user_ids']] for res in users_by_task_read_group}
+        for task in self:
+            task.portal_user_names = ', '.join(assignees_name_by_task.get(task.id, []))
```

Comment:
For the last suggestion, this would give:
```python
        # fetch 'user_ids' in superuser mode (and override value in cache)
        self._read(['user_ids'])
        for task in self.with_context(prefetch_fields=False):
            task.portal_user_names = ', '.join(task.user_ids.mapped('name'))
```

Reply:
Oh so cool, thanks for the explanation and the suggestion ! 🙂